src/charm.py

```diff
@@ -353,6 +383,66 @@ def _config_file_content_matches(self, content: str) -> bool:
         existing_content = self._container.pull(path=f"{BASE_CONFIG_PATH}/{CONFIG_FILE_NAME}")
         return existing_content.read() == content
 
+    def _on_get_profile_a_home_network_public_key_action(self, event: ActionEvent) -> None:
+        if not self._container.can_connect():
+            event.fail(message="Container is not ready yet.")
+            return
+        if not self._profile_a_home_network_private_key_stored():
+            event.fail(message="Home network private key is not stored yet.")
+            return
+        event.set_results(
+            {
+                "public-key": self._get_profile_a_home_network_public_key(),  # type: ignore[arg-type] # noqa: E501
+            }
+        )
+
+    def _generate_profile_a_home_network_private_key(self) -> None:
+        """Generates and stores profile A Home Network private key on the container."""
+        private_key = X25519PrivateKey.generate()
+        private_bytes = private_key.private_bytes(
+            encoding=serialization.Encoding.Raw,
+            format=serialization.PrivateFormat.Raw,
+            encryption_algorithm=serialization.NoEncryption(),
+        )
+        private_key_string = private_bytes.hex()
```

Comment:
This can probably be its own file outside of `charm.py` 

```python
from cryptography.hazmat.primitives import serialization
from cryptography.hazmat.primitives.asymmetric.x25519 import X25519PrivateKey

def generate_x25519_private_key() -> str:
    private_key = X25519PrivateKey.generate()
    private_bytes = private_key.private_bytes(
        encoding=serialization.Encoding.Raw,
        format=serialization.PrivateFormat.Raw,
        encryption_algorithm=serialization.NoEncryption(),
    )
    return private_bytes.hex()
```